src/pal/src/configure.cmake

```diff
@@ -16,6 +16,14 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin AND NOT CMAKE_SYSTEM_NAME STREQUAL Free
   set(CMAKE_REQUIRED_DEFINITIONS "-D_BSD_SOURCE -D_SVID_SOURCE -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L")
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL Linux AND NOT CLR_CMAKE_PLATFORM_ANDROID)
```

Comment:
This is the same set of conditions under which we include `rt` for `target_link_libraries` in https://github.com/dotnet/coreclr/blob/master/src/pal/src/CMakeLists.txt

Reply:
I don't see why clock_gettime would require linking to librt. The man page doesn't mention that.

Reply:
This function is in glibc.

Reply:
It is only applicable to older versions of `glibc`. RHEL explicitly calls it out as a requirement under https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/7/html/reference_guide/sect-posix_clocks (and other similar pages for their other distros, they don't look to have an explicit page for regular RHEL 6).

There are also multiple hits on StackOverflow (and other sites) indicating the same.

Reply:
Ah, interesting.

Reply:
This is explicitly called out in the NEWS section for `glibc` 2.17: https://sourceware.org/git/?p=glibc.git;a=blob;f=NEWS;h=41cffa425c70d5c9cc9500e866bd6b3c6b6377eb;hb=HEAD#l3378

> The 'clock_*' suite of functions (declared in <time.h>) is now available directly in the main C library. Previously it was necessary to link with -lrt to use these functions. This change has the effect that a single-threaded program that uses a function such as `clock_gettime' (and is not linked with -lrt) will no longer implicitly load the pthreads library at runtime and so will not suffer the overheads associated with multi-thread support in other code such as the C++ runtime library.

Reply:
Thank you for the details!

Reply:
(might be worth an up-for-grabs issue to check if RHEL, or other distros, have more differences like this that might impact the "fallback" paths we are supporting).